pyomo/contrib/mindtpy/util.py

```diff
@@ -34,15 +34,6 @@ def model_is_valid(solve_data, config):
     m = solve_data.working_model
     MindtPy = m.MindtPy_utils
 
-    # Check for any integer variables
-    if any(True for v in m.component_data_objects(
-            ctype=Var, descend_into=True)
-            if v.is_integer() and not v.fixed):
-        raise ValueError('Model contains unfixed integer variables. '
-                         'MindtPy does not currently support solution of '
-                         'such problems.')
-        # TODO add in the reformulation using base 2
-
```

Comment:
With this deletion, you might be able to just use the `model_is_valid()` function from `gdpopt.util`. Only difference is that I don't set up the `dual` Suffix here. You'll need to change the `gdpopt.util.model_is_valid()` function slightly to use the `util_blk = getattr(m, solve_data.util_block_name)` idiom instead of directly hardcoding `GDPopt_utils`.

Reply:
Uhm I'm not sure if we agree on the function of this code.
The way it was written before, the model was checked if it contained any integers (since it didn't support integer variables in the solver).
Now I used the `TransformationFactory('contrib.integer_to_binary')` so that models with integer variables can be solved too. Thus, we don't need to check for this anymore.

Reply:
I think there's a misunderstanding here. I was talking about `gdpout.util.model_is_valid()`, which last time I looked does not include any check for integer variables. I am saying that because of the proposed change here, which I agree with, there is a potential opportunity for code deduplication.